dask/dataframe/core.py

```diff
@@ -1365,14 +1424,13 @@ def query(self, expr, **kwargs):
             dsk = dict(((name, i), (pd.DataFrame.query, (self._name, i), expr))
                        for i in range(self.npartitions))
 
+        dummy = self._pd.query(expr, **kwargs)
         return self._constructor(merge(dsk, self.dask), name,
-                                       self.columns, self.divisions)
+                                 dummy, self.divisions)
 
     @derived_from(pd.DataFrame)
     def dropna(self, how='any', subset=None):
-        def f(df, how=how, subset=subset):
-            return df.dropna(how=how, subset=subset)
-        return map_partitions(f, self.columns, self)
+        return self.map_partitions(pd.DataFrame.dropna, how=how, subset=subset)
```

Comment:
In particular, this change will cause `dropna` to fail when using `distributed` as the scheduler. 

I need to find a good way to test for this within the dask codebase.